frontend/src/components/App/Settings/DrawerModeSettings.tsx

```diff
@@ -0,0 +1,53 @@
+import { FormControlLabel, Switch } from '@mui/material';
+import React from 'react';
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setDetailDrawerEnabled } from '../../../redux/drawerModeSlice';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import { TooltipIcon } from '../../common';
+
+export default function DrawerModeSettings() {
+  const dispatch = useDispatch();
+  const { t } = useTranslation('translation');
+
+  const isDrawerEnabled = useTypedSelector(state => state.drawerMode.isDetailDrawerEnabled);
+
+  if (isDrawerEnabled) {
```

Comment:
this part of the code takes isDrawerEnabled from the store and then immediately sets it back to the store, it's not needed. the same thing with the useEffect below

Reply:
done!